distributed/scheduler.py

```diff
@@ -1063,6 +1063,12 @@ class TaskGroup:
     #: Cumulative duration of all completed actions, by action
     all_durations: defaultdict[str, float]
 
+    #: Span ID (see distributed.spans).
+    #: It is possible to end up in situation where different tasks of the same TaskGroup
```

Comment:
Conceptually, I feel like this shouldn't occur. I hope that we can avoid this scenario (or task groups altogether) with `dask-expr` in the future.